sklearn/preprocessing/data.py

```diff
@@ -397,6 +399,239 @@ def inverse_transform(self, X, copy=None):
+        not a NumPy array or scipy.sparse CSR matrix, a copy may still be
+        returned.
+
+    Attributes
+    ----------
+    `center_` : array of floats
+        The median value for each feature in the training set, unless axis=1,
+        in which case it contains the median value for each sample
+
+    `scale_` : array of floats
+        The (scaled) interquartile range for each feature in the training set,
+        unless axis=1, in which case it contains the median value for each
+        sample.
+
+    See also
+    --------
+    :class:`sklearn.preprocessing.StandardScaler` to perform centering
+    and scaling using mean and variance.
+
+    :class:`sklearn.decomposition.RandomizedPCA` with `whiten=True`
+    to further remove the linear correlation across features.
+
+    Notes
+    -----
+    See examples/preprocessing/plot_robust_scaling.py for an example.
+
+    http://en.wikipedia.org/wiki/Median_(statistics)
+    http://en.wikipedia.org/wiki/Interquartile_range
+    """
+
+    def __init__(self, with_centering=True, with_scaling=True, copy=True):
+        self.with_centering = with_centering
+        self.with_scaling = with_scaling
+        self.copy = copy
+
+    def _check_array(self, X, copy):
+        """Makes sure centering is not enabled for sparse matrices."""
+        X = check_array(X, accept_sparse=('csr', 'csc'), dtype=np.float,
+                        copy=copy, ensure_2d=False)
+        if sparse.issparse(X):
+            if self.with_centering:
+                raise ValueError(
+                    "Cannot center sparse matrices: use `with_centering=False`"
+                    " instead. See docstring for motivation and alternatives.")
+        return X
+
+    def _handle_zeros_in_scale(self, scale):
+        ''' Makes sure that whenever scale is zero, we handle it correctly.
+
+        This happens in most scalers when we have constant features.'''
+        # if we are fitting on 1D arrays, scale might be a scalar
+        if np.isscalar(scale):
+            if scale == 0:
+                scale = 1.
+        elif isinstance(scale, np.ndarray):
+            scale[scale == 0.0] = 1.0
+            scale[-np.isfinite(scale)] = 1.0
+        return scale
+
+    def fit(self, X, y=None):
+        """Compute the median and quantiles to be used for scaling.
+
+        Parameters
+        ----------
+        X : array-like with shape [n_samples, n_features]
+            The data used to compute the median and quantiles
+            used for later scaling along the features axis.
+        """
+        if sparse.issparse(X):
+            raise TypeError("RobustScaler cannot be fitted on sparse inputs")
+
+        X = self._check_array(X, self.copy)
+        if self.with_centering:
+            self.center_ = np.median(X, axis=0)
+
+        if self.with_scaling:
+            q = np.percentile(X, (25, 75), axis=0)
+            self.scale_ = (q[1] - q[0])
+            if np.isscalar(self.scale_):
```

Comment:
This can happen if the input isn't actually "array-like with shape [n_samples, n_features]", but rather is a 1D array. `StandardScaler` behaves the same way, and uses the same docstring to describe X. The right thing to do would probably be to change the documentation of both StandardScaler and RobustScaler to somethign like "array-like with shape [n_samples, n_features] or 1D array". Is that a good formulation for it?